### PR TITLE
cmake: Bump minimal cmake version

### PR DESCRIPTION
**Pull request by rleon**
The rdma-core doesn't really need anything specific from 2.8.11, so
simply raise the minimal level to be 2.8.12 and silence the warning
below.

 CMake Deprecation Warning at CMakeLists.txt:7 (cmake_minimum_required):
   Compatibility with CMake < 2.8.12 will be removed from a future version of
   CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

Signed-off-by: Leon Romanovsky <leonro@nvidia.com>

### CODE REVIEW

**Comment by jgunthorpe** [comment]
Woah, IIRC, 2.8.11 was selected because it is what Centos6 ships.

**Comment by rleon** [comment]
This is time to drop centos, most likely FC34 will have incompatible cmake.

**Comment by rleon** [comment]
So, what do we plan to do?

**Comment by jgunthorpe** [comment]
We don't actually need compat with 2.8.11, we don't rely on any special behaviors. Can we just conditionalize it based on the cmake version?

**Comment by rleon** [comment]
> We don't actually need compat with 2.8.11, we don't rely on any special behaviors. Can we just conditionalize it based on the cmake version?

It will look strange, but doable.

**Comment by rleon** [comment]
> > We don't actually need compat with 2.8.11, we don't rely on any special behaviors. Can we just conditionalize it based on the cmake version?
> 
> It will look strange, but doable.

done

**Comment by jgunthorpe** [comment]
Why doesn't it build?

**Comment by rleon** [comment]
I saw the failure but didn't have time to look yet.

**Comment by rleon** [comment]
> It would be nice if we could find a better cmake version that just .12 - check cbuild's matrix and pick the next lowest?

Centos7 uses 2.8.12.2

**Comment by rleon** [comment]
Repushed to check CI

**Comment by rleon** [comment]
Finally passed

File: libibnetdisc/CMakeLists.txt [hunk 1]
```diff
@@ -20,5 +20,6 @@ rdma_pkg_config("ibnetdisc" "libibumad libibmad" "")
 rdma_test_executable(testleaks tests/testleaks.c)
 target_link_libraries(testleaks LINK_PRIVATE
   ibmad
+  ibumad
```
Comment (jgunthorpe) [inline libibnetdisc/CMakeLists.txt:23]: This isn't how things are supposed to work, ibmad shuld be self contained

target_link_libraries(ibmad LINK_PRIVATE
  ibumad
  )

I assume changing the cmake version also changed how some of this stuff works?

File: infiniband-diags/CMakeLists.txt [hunk 2]
```diff
@@ -8,16 +8,13 @@ install(FILES
   etc/ibdiag.conf
   DESTINATION "${IBDIAG_CONFIG_PATH}")
 
-add_library(ibdiags_tools STATIC
```
Comment (jgunthorpe) [inline infiniband-diags/CMakeLists.txt:11]: The library prevents multiple recompilation, why can't we fix the library?

**Review by jgunthorpe — CHANGES_REQUESTED** [review]

File: libibnetdisc/CMakeLists.txt (hunk 1 above)
Comment (rleon) [reply libibnetdisc/CMakeLists.txt:23]: Yes, for some reason SuSE 15.0 didn't build without this change.
See first commit for an error.

File: infiniband-diags/CMakeLists.txt (hunk 2 above)
Comment (rleon) [reply infiniband-diags/CMakeLists.txt:11]: It doesn't worth, I see "fixing" library as optimization of something that no one cares.

File: libibnetdisc/CMakeLists.txt (hunk 1 above)
Comment (rleon) [reply libibnetdisc/CMakeLists.txt:23]: ok, I found the reason to missing -rpath error.
It was caused by CMP0022
https://cmake.org/cmake/help/latest/policy/CMP0022.html
I tried as a workaround the following change
cmake_policy(SET CMP0022 OLD)

and suse built correctly.

Comment (jgunthorpe) [reply libibnetdisc/CMakeLists.txt:23]: That makes alot more sense

Comment (jgunthorpe) [reply libibnetdisc/CMakeLists.txt:23]: But should we make the update described in that policy document, whatever it is?

Comment (rleon) [reply libibnetdisc/CMakeLists.txt:23]: Yes, we need, the cmake_policy() was simplest way to check that we are hitting this specific policy change.

File: infiniband-diags/CMakeLists.txt (hunk 2 above)
Comment (rleon) [reply infiniband-diags/CMakeLists.txt:11]: BTW, I fixed it locall by putting ibdiag_tools as first in the library list.

Just can't push till we fix the suse failure.

**Pull request merged**